app/javascript/src/components/Player.tsx

```diff
@@ -223,102 +225,118 @@ class Player extends React.Component<PlayerProps, PlayerState> {
   }
 
   private renderDebugger() {
-    return this.props.debug ? (
-      <div>
-        <div className="d-helper"></div>
-        <div className="debugger">
-          <div className="d-section">
-            <h4><u>Current Items</u></h4>
-            <ul className="d-list">
-              {this.state.currentItems.map((item, i) => {
-                return (
-                  <li key={item.concat(i.toString())}>
-                    <div>
-                      <label
-                        className="sr-only"
-                        htmlFor={`name-for-${item}`}
-                      >
-                        Item Name
-                      </label>
-                      <select
-                        id={`name-for-${item}`}
-                        value={item}
-                        onChange={this.setItem.bind(this, i)}
-                      >
-                        {this.possibleItems(item).map(
-                          (item, i) => (
-                            <option key={i} value={item}>
-                              {item}
-                            </option>
-                          )
-                        )}
-                      </select>
-                    </div>
-                    <DebuggerRemoveButton
-                      onClick={this.removeItem.bind(this, i)}
-                    />
-                  </li>
-                )
-              })}
-            </ul>
-            <DebuggerFooter onClick={this.addItem.bind(this)} />
-          </div>
-          <div className="d-section">
-            <h4><u>Current Stats</u></h4>
-            <ul className="d-list">
-              {this.state.currentStats.map((stat, i) => {
-                return (
-                  <li key={stat.name.concat(i.toString())}>
-                    <div>
-                      <label
-                        className="sr-only"
-                        htmlFor={`stat-name-${stat.name}`}
-                      >
-                        Stat Name
-                      </label>
-                      <select
-                        id={`stat-name-${stat.name}`}
-                        value={stat.name}
-                        onChange={this.setStatName.bind(this, i)}
-                      >
-                        {this.possibleStats(stat.name).map(
-                          (item, i) => (
-                            <option key={i} value={item}>
-                              {item}
-                            </option>
-                          )
-                        )}
-                      </select>
-                    </div>
-                    <div className="d-stat-edit">
+    if (this.props.debuggable) {
+      return this.debug ? (
+        <div>
```

Comment:
React isn't exactly my area of expertise, but I think it's pretty common practice to pull out the values of big conditionals like this into their own functions. That way your ternary statement becomes nice and compact, and it's easier to see where one block ends and another begins.